Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -3201,6 +3199,48 @@ - (void)showAttachmentDocumentPicker
     [self presentViewController:menuController animated:YES completion:nil];
 }
 
+#pragma mark - Attachment Picking: GIFs
+
+- (void)showGifPicker
+{
+    GifPickerViewController *view =
+        [[GifPickerViewController alloc] initWithThread:self.thread messageSender:self.messageSender];
+    view.delegate = self;
+    UINavigationController *navigationController = [[UINavigationController alloc] initWithRootViewController:view];
+    [self presentViewController:navigationController animated:YES completion:nil];
+}
+
+#pragma mark GifPickerViewControllerDelegate
+
+- (void)gifPickerWillSend
+{
+    [ThreadUtil addThreadToProfileWhitelistIfEmptyContactThread:self.thread];
+}
+
+- (void)gifPickerDidSendWithOutgoingMessage:(TSOutgoingMessage *)message
+{
+    [self messageWasSent:message];
+
+    [self ensureDynamicInteractions];
+}
+
+- (void)messageWasSent:(TSOutgoingMessage *)message
```

Comment:
This method should to be called after we send any text or attachment message from conversation view.

Reply:
👍